libs/datamodel/core/src/transform/ast_to_dml/generator_loader.rs

```diff
@@ -71,10 +96,18 @@ impl GeneratorLoader {
         };
 
         if !preview_features.is_empty() {
-            if let Err(err) =
-                validate_preview_features(preview_features.clone(), span, Vec::from(GENERATOR_PREVIEW_FEATURES))
-            {
-                return Err(err);
+            let mut result = validate_preview_features(
+                preview_features.clone(),
+                span,
+                Vec::from(GENERATOR_PREVIEW_FEATURES),
+                Vec::from(DEPRECATED_GENERATOR_PREVIEW_FEATURES),
+            );
+            if result.has_errors() {
+                diagnostics.append(&mut result);
+                return Err(diagnostics);
+            }
```

Comment:
There are similar patterns like that one in the PR. Wonder whether those can't be dried up. Maybe something like that in order to turn this into a common one liner:
```rust
diagnostic.merge(result).as_error()?
```